File: src/main/scala/com/rohitpothuri/rbac/workflows/GreetingWorkflow.scala
package com.rohitpothuri.rbac.workflows

import io.temporal.workflow.{WorkflowInterface, WorkflowMethod}

@WorkflowInterface
trait GreetingWorkflow {
  @WorkflowMethod def execute(name: String): String
  /* @WorkflowMethod
      Promise<String> executeAsync(String name);*/
}

File: src/main/scala/com/rohitpothuri/rbac/workflows/GreetingWorkflow.java
package com.rohitpothuri.rbac.workflows;

import io.temporal.workflow.Promise;
import io.temporal.workflow.WorkflowInterface;
import io.temporal.workflow.WorkflowMethod;

@WorkflowInterface
public interface GreetingWorkflow {

    @WorkflowMethod
    String execute(String name);

   /* @WorkflowMethod
    Promise<String> executeAsync(String name);*/
}